src/lib.rs

```diff
@@ -19,12 +19,23 @@ pub mod worktree;
 /// The bool in the return value specifies whether there is a repository
 /// in root itself.
 #[allow(clippy::type_complexity)]
-fn find_repos(root: &Path) -> Result<Option<(Vec<repo::Repo>, Vec<String>, bool)>, String> {
+fn find_repos(
+    root: &Path,
+    exclusion_pattern: &Option<String>,
```

Comment:
Instead of taking a reference to an option, it's a bit more ergonomic to instead take an option that **contains** a reference. Instead of `&Option<T>`, you'd generally use `Option<&T>`.

And because we are handling strings here, instead of passing `&String`, you'd rather pass a slice (`&str`). That makes it easier for the caller to pass strings as-is (e.g. `find_repos(foo, Some("myregex"))` without further mangling. The caller never needs actual ownership of the `String`.

This also means that you can get rid of the `clone()` calls in line 31, as you are only passing string slices (i.e. references) around.

Note that this is completely irrelevant for the logic of the code in this case. It's just something good to know!

```suggestion
    exclusion_pattern: Option<&str>,
```